packages/components/src/SquigglePlayground.tsx

```diff
@@ -0,0 +1,129 @@
+            Props.onChange(result);
+          }
+        }}
+      />
+    </Form.Item>
+  );
+}
+
+interface Props {
+  initialSquiggleString: string;
+}
+
+let SquigglePlayground: FC<Props> = (props) => {
+  let [squiggleString, setSquiggleString] = useState(
+    props.initialSquiggleString
+  );
+  let [sampleCount, setSampleCount] = useState(1000);
+  let [outputXYPoints, setOutputXYPoints] = useState(1000);
+  let [pointDistLength, setPointDistLength] = useState(1000);
+  let [diagramStart, setDiagramStart] = useState(0);
+  let [diagramStop, setDiagramStop] = useState(10);
+  let [diagramCount, setDiagramCount] = useState(20);
+  var demoDist = (
+    <SquiggleChart
+      squiggleString={squiggleString}
+      sampleCount={sampleCount}
+      outputXYPoints={outputXYPoints}
+      diagramStart={diagramStart}
+      diagramStop={diagramStop}
+      diagramCount={diagramCount}
+      pointDistLength={pointDistLength}
+    />
+  );
+  return (
+    <Row>
+      <Col span={12}>
+        <Card title="Distribution Form">
+          <Form>
+            <Row gutter={16}>
+              <Col span={24}>
+                <CodeEditor
+                  value={squiggleString}
+                  onChange={setSquiggleString}
+                  oneLine={false}
+                />{" "}
+              </Col>
+            </Row>
+            <Row gutter={16}>
+              <Col span={12}>
+                <FieldFloat
+                  value={sampleCount}
+                  label="Sample Count"
+                  onChange={setSampleCount}
+                />{" "}
+              </Col>
+              <Col span={12}>
+                <FieldFloat
+                  value={outputXYPoints}
+                  onChange={setOutputXYPoints}
+                  label="Output XY-points"
+                />
+              </Col>
+              <Col span={12}>
+                <FieldFloat
+                  value={pointDistLength}
+                  onChange={setPointDistLength}
+                  label="Downsample To"
+                />
+              </Col>
+              <Col span={12}>
+                <FieldFloat
+                  value={diagramStart}
+                  onChange={setDiagramStart}
+                  label="Diagram Start"
+                />
+              </Col>
+              <Col span={12}>
+                <FieldFloat
+                  value={diagramStop}
+                  onChange={setDiagramStop}
+                  label="Diagram Stop"
+                />{" "}
+              </Col>
+              <Col span={12}>
+                <FieldFloat
+                  value={diagramCount}
+                  onChange={setDiagramCount}
+                  label="Diagram Count"
+                />
+              </Col>
+            </Row>
+          </Form>
+        </Card>
+      </Col>
+      <Col span={12}>{demoDist}</Col>
+    </Row>
+  );
+};
+export default SquigglePlayground;
+export function renderSquigglePlayground(props: Props) {
```

Comment:
This function is pretty weird to me.

It seems very generic to the situation (just creating a document and rendering the component). I'd just have a function called ``renderComponent`` or something, outside of this file, and pass it the react component, or similar.

Reply:
I'm just thinking about the user experience from the observable side. With this function, it's pretty simple to create a playground without pulling in any more dependencies.

Because observable is a pretty baron environment, I kind of don't want the user to think too much about whether it's a React component or not.

I was thinking in the past to do this, but in the end I didn't think we needed that flexibility to render _any_ react component

Reply:
How about you rename it to renderSquigglePlaygroundToDom? 

I think this is still a pretty unusual sort of function, and it strikes me as kind of inelegant because of this. (In terms of the interface). But I'm not going to fight it much, especially if it has a name that makes this more obvious.

Reply:
I've renamed it to make it a bit more obvious. I know it's ugly, but I feel like from an interface perspective, particularly from observable, the alternatives are uglier.